### PR TITLE
Merge runtime builtin scripts.

### PR DESCRIPTION
**Pull request by dumganhar**

### CODE REVIEW

File: cocos2d/core/platform/instantiate-jit.js [hunk 1]
```diff
@@ -162,7 +162,14 @@ function equalsToDefault (def, value) {
             (def.constructor === Object && value.constructor === Object)
         ) {
             try {
-                return JSON.stringify(def) === JSON.stringify(value);
+                if (CC_JSB) {
+                    //FIXME: JSON.stringify will take lots of time for compare object at runtime on JSB.
+                    // We need to consider a new way for checking whether two objects are equal.
```
Comment (dumganhar) [inline cocos2d/core/platform/instantiate-jit.js]: @jareguo needs to fix this.

**Review by pandamicro — APPROVED** [review]

File: cocos2d/core/platform/instantiate-jit.js (hunk 1 above)
Comment (jareguo) [reply cocos2d/core/platform/instantiate-jit.js]: Okey, then just use
```
return Array.isArray(def) && Array.isArray(value) && def.length === 0 && value.length === 0;
```
in line #161:

Comment (dumganhar) [reply cocos2d/core/platform/instantiate-jit.js]: But this is only for checking whether the length of two arrays is equal.
How about two array length is equal but their contents are not?

Comment (jareguo) [reply cocos2d/core/platform/instantiate-jit.js]: so length should be 0

**Pull request merged**